python/cudf/cudf/utils/ioutils.py

```diff
@@ -1220,6 +1229,120 @@ def _get_filesystem_and_paths(path_or_data, **kwargs):
     return fs, return_paths
 
 
+def _set_context(obj, stack):
+    # Helper function to place open file on context stack
+    if stack is None:
+        return obj
+    return stack.enter_context(obj)
+
+
+def _open_remote_files(
+    paths,
+    fs,
+    context_stack=None,
+    open_file_func=None,
+    precache_options=None,
+    **kwargs,
+):
+    """Return a list of open file-like objects given
+    a list of remote file paths.
+
+    WARNING: This utility is currently experimental,
+    and is meant for internal cudf use only.
```

Comment:
I don't think this is necessary, by convention we just let the `_` denote internal utilities users shouldn't rely on.

Reply:
Good point - I added this warning when I was considering it as a public function (and when it was called `open_remote_files`)